aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/UpdateHandler.java

```diff
@@ -80,8 +80,12 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
                         () -> modifyDBCluster(proxy, rdsProxyClient, progress, previousResourceState, desiredResourceState, isRollback),
                         CallbackContext::isModified,
                         CallbackContext::setModified))
-                .then(progress -> removeAssociatedRoles(proxy, rdsProxyClient, progress, setDefaults(request.getPreviousResourceState()).getAssociatedRoles()))
-                .then(progress -> addAssociatedRoles(proxy, rdsProxyClient, progress, progress.getResourceModel().getAssociatedRoles()))
+                .then(progress -> updateAssociatedRoles(
+                        proxy,
+                        rdsProxyClient,
+                        progress,
+                        setDefaults(request.getPreviousResourceState()).getAssociatedRoles(),
```

Comment:
I believe this mutating call could cause unexpected side effects here: the `setDefaults` has an effect on the entire previous resource state. Passing `request.getPreviousResourceState.getAssociatedRoles()` should be enough.

updateAssociatedRoles should be able to handle nullable collections. This would be enough.

Reply:
Done. Thanks Oleg!